### PR TITLE
Support loading encondings from a GlyphOrderAndAliasDB file

### PR DESCRIPTION
**Pull request by felipesanches**
Also did some cleanup and refactoring on the functions related to this portion of the code.

### CODE REVIEW

**Comment by davelab6** [comment]
AWESOME :D Thanks Felipe! I hope @frank-trampe can take a look soon, once he has wrapped up everything needed for the August release :)

**Comment by frank-trampe** [comment]
@felipesanches, thanks for doing this! Could you submit a pull request without the whitespace changes?

**Comment by felipesanches** [comment]
Sorry, but the refactoring work was done in order to get the code in a good state before I could actually understand what it does and implement the feature on top of it. Undoing that would be contra-productive work, making the code worse and more error prone.

**Comment by frank-trampe** [comment]
Okay. I'll try to dig through the full pull request when I get the chance then.

**Comment by felipesanches** [comment]
thanks a lot! :-)

File: fontforge/encoding.c [hunk 1]
```diff
@@ -548,38 +548,103 @@ static Encoding *ParseConsortiumEncodingFile(FILE *file) {
...
+            any = 1;
+        } else {
+            names[i] = copy(&buffer[tab+1]);
+            any = 1;
+        }
+        i++;
+    }
+
+    if (max != -1) {
+        if (++max < 256) max = 256;
```
Comment (frank-trampe) [inline fontforge/encoding.c]: Where do we get the minimum count?

File: fontforge/encoding.c [hunk 2]
```diff
@@ -548,38 +548,103 @@ static Encoding *ParseConsortiumEncodingFile(FILE *file) {
...
+            buffer[tab2]='\0';
+
+        if (strcmp(buffer, ".notdef") == 0) {
+            encs[i] = -1;
+        } else if ((enc = UniFromName(buffer, ui_none, &custom)) != -1) {
+            encs[i] = enc;
+
+            /* Used not to do this, but there are several legal names */
+            /*  for some slots and people get unhappy (rightly) if we */
+            /*  use the wrong one */
```
Comment (frank-trampe) [inline fontforge/encoding.c]: We were not doing any of this before, so I'm not sure what this comment references. Are you referring to trying to look up the Unicode name?

**Comment by davelab6** [comment]
@felipesanches ping on frank's comments

**Comment by davelab6** [comment]
@felipesanches ping

**Comment by felipesanches** [comment]
hi, @frank-trampe 
I would like to get this code merged. I can dedicate some time to review the code in this pull request and possibly improve it today (or later this week). Do you see any blocking issue with the current code? Anything you'd say is mandatory before this gets finally merged?

@davelab6 you're still interested in getting this merged into master, right?

**Comment by frank-trampe** [comment]
@felipesanches, I'd love to get this merged, but it's a bogglingly large pull request, and filtering out whitespace changes doesn't shrink it very much. Are you sure you can't remember where you made actual changes? And when might you have a chance to answer my questions? Once I have all of the information, I can probably rebuild this into a compact single-commit pull request and get it merged, which would be a very good thing.

**Comment by felipesanches** [comment]
Alright. I'll clean it up trying to respect the legacy ugly-spacing policy so that the commits are minimal and you have less code to actually review.

Thanks for the very quick reply ;-)

**Comment by davelab6** [comment]
Yes, would be good to get this merged.

**Comment by felipesanches** [comment]
There's a new pull request now at https://github.com/fontforge/fontforge/pull/2702

I'll close this one as it is now deprecated.

**Pull request closed without merging**